### PR TITLE
Fix some gir annotations

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Based on https://github.com/gtk-rs/gtk3-rs/pull/548.

### CODE REVIEW

File: gtk/src/auto/functions.rs [hunk 1]
```diff
@@ -135,11 +135,6 @@ pub fn accelerator_parse(accelerator: &str) -> (u32, gdk::ModifierType) {
     }
 }
 
-//#[doc(alias = "gtk_accelerator_parse_with_keycode")]
-//pub fn accelerator_parse_with_keycode(accelerator: &str, accelerator_codes: Vec<u32>) -> (u32, gdk::ModifierType) {
```
Comment (bilelmoussaoui) [inline gtk/src/auto/functions.rs:139]: you can copy this one from gtk4 if you want :)

File: atk/Gir.toml [hunk 2]
```diff
@@ -67,6 +66,13 @@ name = "Atk.Action"
 status = "generate"
 trait_name = "AtkActionExt" # duplicate with gio
 
+[[object]]
+name = "Atk.Document"
+status = "generate"
+    [[object.function]]
+    name = "get_attributes"
+    ignore = true # TODO: re-enable once AttributeSet is properly handled
```
Comment (sdroege) [inline atk/Gir.toml:74]: Why do you ignore them all if they can be implemented? Just leave them around, then at least we are aware that they still need to be bound if someone cares enough (and they should up in the "unbound" mode of gir)

Comment (bilelmoussaoui) [reply atk/Gir.toml:74]: Because of the docs, it will create broken links otherwise

Comment (sdroege) [reply atk/Gir.toml:74]: Annoying

Comment (bilelmoussaoui) [reply atk/Gir.toml:74]: Ideally gir would mark those functions/types as not possible to bound so we can ignore them when generating docs links

Comment (bilelmoussaoui) [reply atk/Gir.toml:74]: Currently they keep their generate state

File: gtk/src/auto/functions.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply gtk/src/auto/functions.rs:139]: Done, thanks! :)

File: gtk/src/functions.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            loop {
+                // zero-terminated array
+                let ret = std::ptr::read(accelerator_codes_ptr.add(i));
+                if ret == 0 {
+                    break;
+                }
+                accelerator_codes.push(ret);
+                i += 1;
+            }
+            glib::ffi::g_free(accelerator_codes_ptr as *mut _);
```
Comment (sdroege) [inline gtk/src/functions.rs:33]: If you feel like it

```rust
let mut len = 0;
if !accelerator_codes_ptr.is_null() {
    while ptr::read(accelerator_codes_ptr.add(len)) != 0 {
       len += 1;
    }
}
let accelerator_codes = glib::GSlice::from_glib_full_num(accelerator_codes_ptr, len);
```

That way you don't have to unnecessarily allocate twice but can re-use the allocation from GTK here.

Comment (GuillaumeGomez) [reply gtk/src/functions.rs:33]: :o

**Review by sdroege — APPROVED** [review]
Good for me otherwise

File: gtk/src/functions.rs (hunk 3 above)
Comment (sdroege) [reply gtk/src/functions.rs:33]: Well, that's why I started adding `glib::GSlice` and things like that. I have some WIP changes locally for more things and having gir autogenerate that (in other cases, not here, this function is complicated).

Comment (GuillaumeGomez) [reply gtk/src/functions.rs]: Done!

**Pull request merged**